econml/cate_estimator.py

```diff
@@ -141,23 +118,37 @@ def marginal_effect(self, T, X=None):
         """
         pass
 
+    def _defer_to_inference(m):
+        @wraps(m)
+        def call(self, *args, **kwargs):
+            name = m.__name__
+            if self._inference is not None:
+                return getattr(self._inference, name)(*args, **kwargs)
+            else:
+                AttributeError("Can't call '%s' because 'inference' is None" % name)
+        return call
+
+    @_defer_to_inference
+    def effect_interval(self, X=None, *, T0=0, T1=1, alpha=0.1):
+        pass
+
+    @_defer_to_inference
+    def marginal_effect_interval(self, T, X=None, *, alpha=0.1):
+        pass
+
 
 class LinearCateEstimator(BaseCateEstimator):
     """
     Base class for all CATE estimators with linear treatment effects in this package.
 
     Parameters
     ----------
-    inference: string, inference method, or None
+    inference: string, `Inference` instance, or None
         Method for performing inference.  All estimators support 'bootstrap'
-        (or an instance of `BootstrapOptions`), some support other methods as well.
+        (or an instance of `BootstrapInference`), some support other methods as well.
 
     """
 
-    @abc.abstractmethod
-    def __init__(self, inference):
-        super().__init__(inference=inference)
-
     @abc.abstractmethod
     def const_marginal_effect(self, X=None):
```

Comment:
When T0, T1 are scalars, we should copy this scalar to all the treatments. So we expand them to 2D arrays if we are in a multi-dimensional treatment case. This would alleviate one problem I was getting, when calling effect_interval with the default T0, T1 and a multi-dim treatment.

Reply:
This is now handled appropriately.